openreview/tools.py

```diff
@@ -194,3 +289,44 @@ def get_paper_conflicts(client, paper):
         domain_conflicts.remove('gmail.com')
 
     return (domain_conflicts, relation_conflicts)
+
+def get_paperhash(first_author, title):
+    title = title.strip()
+    strip_punctuation = '[^A-zÀ-ÿ\d\s]'.decode('utf-8')
+    title = re.sub(strip_punctuation, '', title)
+    first_author = re.sub(strip_punctuation, '', first_author)
+    first_author = first_author.split(' ').pop()
+    title = re.sub(strip_punctuation, '', title)
+    title = re.sub('\r|\n', '', title)
+    title = re.sub('\s+', '_', title)
+    first_author = re.sub(strip_punctuation, '', first_author)
+    return (first_author + '|' + title).lower()
+
+def replace_members_with_ids(client, group):
+    ids = []
+    emails = []
+    for member in group.members:
+        if '~' not in member:
+            try:
+                profile = client.get_profile(member)
+                ids.append(profile.id)
+            except openreview.OpenReviewException as e:
+                if ['Profile not found'] in e:
+                    emails.append(member)
+                else:
+                    raise e
+        else:
+            ids.append(member)
+
+    group.members = ids + emails
+    client.post_group(group)
+
+def get_all_notes(client, invitation, limit=1000):
+    done = False
+    notes = []
+    while not done:
+        batch = client.get_notes(invitation=invitation, limit=limit)
```

Comment:
you need to pass an offset otherwise you get always the same result.

Reply:
huh, you are right, that's weird. the test I did must have been flawed somehow.